app/models/hmcts_common_platform/defendant_summary.rb

```diff
@@ -52,12 +54,34 @@ def offence_summaries
       end
     end
 
+    def application_summaries
+      Array(data[:applicationSummaries]).map do |application_summary|
+        HmctsCommonPlatform::DefendantCourtApplicationSummary.new(application_summary)
+      end
+    end
+
     def to_json(*_args)
       to_builder.attributes!.compact
     end
 
   private
 
+    def match_application_summaries
+      @data[:applicationSummaries] = []
+
+      return unless court_application_summaries.is_a?(Array) && court_application_summaries.any?
+
+      court_application_summaries.each do |summary|
+        master_defendant_id = summary.dig("subjectSummary", "masterDefendantId").to_s.strip
+
+        expected_defendant_id = data[:defendantId].to_s.strip
+
+        next unless master_defendant_id == expected_defendant_id
```

Comment:
You still don't have any actual tests of this functionality. I'd like a test that demonstrates that if the masterDefendantId matches, the summary is included in the payload, and one that shows if it doesn't match, it's not included.